sequencer/src/l1_client.rs

```diff
@@ -140,15 +140,16 @@ impl L1Client {
         prev_finalized: Option<u64>,
         new_finalized: u64,
     ) -> Vec<FeeInfo> {
-        // `prev` should have allready been processed unless we
-        // haven't processed *any* blocks yet.
-        let prev = prev_finalized.map(|prev| prev + 1).unwrap_or(0);
         // No new blocks have been finalized, therefore there are no
         // new deposits.
-        if prev >= new_finalized {
+        if prev_finalized == Some(new_finalized) {
             return vec![];
         }
 
+        // `prev` should have allready been processed unless we
+        // haven't processed *any* blocks yet.
+        let prev = prev_finalized.map(|prev| prev + 1).unwrap_or(0);
+        dbg!(prev, new_finalized);
```

Comment:
We should replace all `dbg!` with `tracing::debug!` or something, as `dbg!` unconditionally outputs regardless of the log level which can make the logging very noisy

Reply:
OH yea I need to be more careful about removing those, maybe I'll add a check to my local hooks. In any case, it's gone now.